cmd/health-monitor.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/spf13/cobra"
+)
+
+type monitorOptions struct {
+	EnableMon         bool     `config:"enable-monitoring"`
+	DisableList       []string `config:"monitor-disable-list"`
+	BfsPollInterval   int      `config:"blobfuse2-poll-interval"`
+	StatsPollinterval int      `config:"stats-poll-interval"`
+}
+
+var pid string
+var cacheMonitorOptions file_cache.FileCacheOptions
+var hmonOptions monitorOptions
+
+func resetMonitorOptions() {
+	hmonOptions = monitorOptions{}
+	cacheMonitorOptions = file_cache.FileCacheOptions{}
+}
+
+var healthMonCmd = &cobra.Command{
+	Use:               "health-monitor",
+	Short:             "Monitor blobfuse2 mount",
+	Long:              "Monitor blobfuse2 mount",
+	SuggestFor:        []string{"healthmon", "monitor health"},
+	Args:              cobra.ExactArgs(0),
+	Hidden:            true,
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		resetMonitorOptions()
+
+		err := validateMonOptions()
+		if err != nil {
+			log.Err("health-monitor: [%v]", err)
+			return err
+		}
+
+		options.ConfigFile = configFile
+		parseConfig()
+
+		err = config.UnmarshalKey("file_cache", &cacheMonitorOptions)
+		if err != nil {
+			log.Err("health-monitor: FileCache config error [invalid config attributes]")
+			return err
+		}
+
+		err = config.UnmarshalKey("health-monitor", &hmonOptions)
+		if err != nil {
+			log.Err("health-monitor: FileCache config error [invalid config attributes]")
+			return err
+		}
+
+		cliParams := buildCliParamForMonitor()
+		log.Debug("health-monitor: options = %v", cliParams)
+
+		log.Debug("Starting health-monitor for blobfuse2 pid = %s", pid)
+
+		hmcmd := exec.Command(hmcommon.HealthMon, cliParams...)
+		cliOut, err := hmcmd.Output()
+		if len(cliOut) > 0 {
+			log.Debug("health-monitor: cliout = %v", string(cliOut))
+			fmt.Println(string(cliOut))
+		}
+		if err != nil {
+			log.Err("health-monitor: [%v]", err)
+			return fmt.Errorf("failed to start health monitor: [%v]", err)
+		}
+
+		return nil
+	},
+}
+
+func validateMonOptions() error {
+	pid = strings.TrimSpace(pid)
+	configFile = strings.TrimSpace(configFile)
+	errMsg := ""
+
+	if len(pid) == 0 {
+		errMsg = "Pid of blobfuse2 process not given\n"
+	}
+
+	if len(configFile) == 0 {
+		errMsg += "Config file not given\n"
+	}
+
+	if len(errMsg) != 0 {
+		errMsg += "Failed to start health-monitor"
+		return fmt.Errorf(errMsg)
+	}
+
+	return nil
+}
+
+func buildCliParamForMonitor() []string {
+	var cliParams []string
+
+	cliParams = append(cliParams, "--pid="+pid)
+	if hmonOptions.BfsPollInterval != 0 {
+		cliParams = append(cliParams, fmt.Sprintf("--blobfuse2-poll-interval=%v", hmonOptions.BfsPollInterval))
+	}
+	if hmonOptions.StatsPollinterval != 0 {
+		cliParams = append(cliParams, fmt.Sprintf("--stats-poll-interval=%v", hmonOptions.StatsPollinterval))
+	}
+	cliParams = append(cliParams, "--cache-path="+cacheMonitorOptions.TmpPath)
```

Comment:
Do we need to sanitize these paths. It may happen that blobfuse failed on mount and statsmon just still kicks in.

Reply:
In this case, even if the monitor is triggered it won't start since the blobfuse2 pid that we passed to the monitor binary is not active.